### ISSUE TITLE
Incorrect information attached to slice::from_ref

### ISSUE THREAD

**Issue by phaazon** [rust-lang/rust#54673]
Head over on the official documentation of [slice::from_ref](https://doc.rust-lang.org/std/slice/fn.from_ref.html). There’s a small information bubble that will trigger the following, confusing information:

![img_092918_180315](https://user-images.githubusercontent.com/506592/46247886-2c826a80-c412-11e8-9360-9e37438f9f8e.png)

I think other functions from the `std` have this problem as well and this might be due to some macro or code generation.

Cheers! :beers:

**Comment by GuillaumeGomez** [comment]
No it's actually expected. However, a bit more explanations could be useful? Any idea @QuietMisdreavus?

**Comment by phaazon** [comment]
What do you mean expected? It’s clearly confusing from a user’s perspective. :)

**Comment by GuillaumeGomez** [comment]
I agree, that's why I asked to @QuietMisdreavus how it could be improved (since @QuietMisdreavus is the one who added this feature iirc).

**Comment by QuietMisdreavus** [comment]
Hmm, the "important traits" box has been mildly confusing for a lot of people since i introduced it...

The feature is intended to be there. The idea behind having that box is that if a function returns an iterator, you can check the type of that iterator without having to load the page for the type.

Imagine a function like [`[T]::windows`](https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.windows). The documentation says it returns "an iterator over all contiguous windows" of the base slice, but how is that represented type-wise? That's the key use-case here:

![image](https://user-images.githubusercontent.com/5217170/47463083-d1425e80-d7aa-11e8-9867-e1e206761c9d.png)

It's an iterator over `&[T]`! And you didn't have to leave the page to see that.

When i opened the PR to introduce it, it was also suggested to include Read and Write on that box, but i feel like those two traits have caused more harm than good. (cc @llogiq, i think you were the one that suggested that.)

So, the function has the info-box because it's returning a (generic) slice, and a (specific) slice implements both Read and Write. Perhaps if we didn't generate this box if the type in question is a primitive? That would also clear up the `&mut T` case showing all three traits.

**Comment by llogiq** [comment]
We should probably distinguish impls that only target a subset of the type (e.g. by trait bound not in the type itself) from unbound impls better.
